testsuite/forge.py

```diff
@@ -1443,19 +1449,23 @@ def test(
         forge_runner = forge_runner_mapping[forge_runner_mode]()
         result = forge_runner.run(context)
 
-        print(result.format())
-
         outputs = []
         if forge_output:
             outputs.append(ForgeFormatter(forge_output, lambda *_: result.output))
         if forge_report:
             outputs.append(ForgeFormatter(forge_report, format_report))
+        else:
+            print(format_report(context, result))
         if forge_comment:
             outputs.append(ForgeFormatter(forge_comment, format_comment))
+        else:
+            print(format_comment(context, result))
```

Comment:
would this just print `format_comment` a whole bunch of times if the test fails?

Reply:
Comment handles failure / success anyway that is what the sticky comment posts, it will not print more or less based on failure / success